src/Sylius/Bundle/CoreBundle/EventListener/Workflow/OrderCheckout/AbstractStateCompletedStateResolverListener.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\EventListener\Workflow\OrderCheckout;
+
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Order\StateResolver\StateResolverInterface;
+use Symfony\Component\Workflow\Event\CompletedEvent;
+use Webmozart\Assert\Assert;
+
+abstract class AbstractStateCompletedStateResolverListener
+{
+    public function __construct(protected StateResolverInterface $stateResolver)
+    {
+    }
+
+    public function onCompleted(CompletedEvent $event): void
```

Comment:
I think it's too restrictive because someone may add a new status and want to move events to another transition, so they'll have to override the entire event.